.github/workflows/book.yml

```diff
@@ -0,0 +1,55 @@
+# Performs checks, builds and deploys the mdbook
+
+name: book
+
+on:
+  push:
+    branches: [main, next]
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup mdbook and extensions
+        run: |
+          # Update Rust and install mdbook, katex, linkcheck and alerts
+          rustup update --no-self-update stable
+          cargo +stable install mdbook mdbook-katex mdbook-linkcheck mdbook-alerts
+
+      - name: Build book
+        run: mdbook build docs
+
+  deploy:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup mdbook and extensions
```

Comment:
It would be nice if we could put this into its own job/file. This is to avoid having a skipped check in every PR, which would make some people like me double check why it was skipped, etc.

![image](https://github.com/user-attachments/assets/9b624710-2226-4b4c-9b53-88accf68a923)

So it would be nice to avoid that bit of mental load :)